src/openforms/submissions/api/fields.py

```diff
@@ -76,3 +78,10 @@ class PrivacyPolicyAcceptedField(BooleanField):
             _("You must accept the privacy policy."),
         )
     ]
+
+
+class FormioDataField(DictField):
+    """A dict field which supports nested data access via dot notation."""
+    def to_internal_value(self, data):
+        super().to_internal_value(data)
+        return FormioData(data)
```

Comment:
```suggestion
    def to_internal_value(self, data):
        data = super().to_internal_value(data)
        return FormioData(data)
```
I'd expect the return value of the super call to be used, no?